sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -527,7 +527,7 @@ public void write(final ChannelHandlerContext context, final Object message, fin
 
         if (message == RntbdHealthCheckRequest.MESSAGE) {
 
-            context.write(RntbdHealthCheckRequest.MESSAGE, promise).addListener(completed -> {
+            context.writeAndFlush(RntbdHealthCheckRequest.MESSAGE, promise).addListener(completed -> {
```

Comment:
What is the reason behind this change ?

Reply:
It is not a required change. But generally write is only queue the message but not really send through wire, writeAndFlush will help flush the message further down the pipeline, so I think it makes sense to keep this change. 

But I am also open to exclude this change for now and may have a PR in future separately for this

Reply:
In any case, there is one more `context.write` we should make sure to flush that one as well. 

If we want to change it, do you know when were we doing the flush in just `write` case ?

Reply:
This potentially can be a perf impacting change.
If we decide to include this we should study the impact of this on both latency and throughput. I think the following studies would be needed:
1. both latency and throughput using the normal benchmark flow
2. both latency and throughput on sparse workload (https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java#L144)

Reply:
I agree, If we want to change it, we should understand how the flushing was working so far and why we want to change it ?
Because I remember in my past conversations with David, flushing immediately had some downsides as well. I can't remember the exact downsides, may be they are not present any more. But a perf check would help us figure it out.

Reply:
@kushagraThapar My current understanding is that, when we call write, the message will be queued in the out buffer, but it is not for sure that the message will be written to the socket. Netty will decide when to flush the message out, so that is why we do not have to call flush in our code. 

@moderakh  Yea, make sense, and those are good tests to do for this change.

@kushagraThapar  @moderakh  I think I will revert this change for now, so this PR only contains style/renaming change,  and I will have a separate PR for this change and to investigate more about it.

Reply:
Makes sense @xinlian12 , thanks.